gprofiler/profilers/perf.py

```diff
@@ -266,44 +267,24 @@ def add_exe_metadata(self, process: Process, metadata: Dict[str, Any]) -> None:
 
 class GolangPerfMetadata(PerfMetadata):
     def relevant_for_process(self, process: Process) -> bool:
-        version = self._get_golang_version(process)
-        return version is not None
-
-    @functools.lru_cache(maxsize=4096)
-    def _get_golang_version(self, process: Process) -> Optional[str]:
-        elf_path = f"/proc/{process.pid}/exe"
-        try:
-            symbol_data = read_elf_symbol(elf_path, "runtime.buildVersion", 16)
-        except FileNotFoundError:
-            raise NoSuchProcess(process.pid)
-        if symbol_data is None:
-            return None
-
-        # Declaration of go string type:
-        # type stringStruct struct {
-        # 	str unsafe.Pointer
-        # 	len int
-        # }
-        addr, length = struct.unpack("QQ", symbol_data)
         try:
-            golang_version_bytes = read_elf_va(elf_path, addr, length)
-        except FileNotFoundError:
-            raise NoSuchProcess(process.pid)
-        if golang_version_bytes is None:
-            return None
-
-        return golang_version_bytes.decode()
+            return is_golang_process(process)
+        except MissingExePath:
+            return False
 
     def make_application_metadata(self, process: Process) -> Dict[str, Any]:
-        metadata = {"golang_version": self._get_golang_version(process)}
+        metadata = {"golang_version": get_process_golang_version(process)}
         self.add_exe_metadata(process, metadata)
         metadata.update(super().make_application_metadata(process))
         return metadata
 
 
 class NodePerfMetadata(PerfMetadata):
     def relevant_for_process(self, process: Process) -> bool:
-        return is_process_basename_matching(process, r"^node$")
+        try:
+            return is_node_process(process)
+        except MissingExePath:
+            return False
```

Comment:
While this `MissingExePath` change is logical, I don't understand why it didn't occur before... Do you?

Reply:
@michelhe is to blame:
https://github.com/Granulate/granulate-utils/pull/73#discussion_r974358851

Reply:
No, that's about golang, not about the nodejs one..

Reply:
כל הזמן מאשימים אותי ורוצים להרוג אותי

![image](https://user-images.githubusercontent.com/2903914/191714122-4fd7a441-7cb0-4b50-87ee-387eae6b5a31.png)

Reply:
You're right.

Reply:
It seems as if the code is literally equal. Can't seem to find the reason.

Reply:
It looks necessary, i'm fine with it

Reply:
Yeah I guess... still weird

Reply:
I actually liked more the fix done in https://github.com/Granulate/gprofiler/pull/451 (avoids calling the function for kernel threads). The same, logically, needs to happen in the callsite for `relevant_for_process`, because we don't collect metadata for kernel threads.

Reply:
@Jongy Can you clarify what needs to be done here? we are doing positive matching (the function in utils does `r"^node"` so how we'll get kernel threads here)

Reply:
Resolving this, talked f2f

Reply:
The exception handling isn't required now.
@Jongy I think the kernel threads check isn't required here.

If you're fine with it, feel free to merge/let me merge.

Reply:
I'm okay with it now, although I don't understand why it's not required. We still use the `is_process_basename_matching` function on kernel threads and it should raise. Anyway.. we'll deal when we see it.